### PR TITLE
Server: Sanitize addresses for all endpoints

### PR DESCRIPTION
**Pull request by facuspagnuolo**

### CODE REVIEW

File: packages/server/src/helpers/sanitize-address-middleware.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+const sanitizeAddresses = object => {
+  Object.keys(object).forEach(key => {
+    if (!!object[key] && typeof object[key] === 'object') sanitizeAddresses(object[key])
+    else if(isAddress(object[key])) object[key] = object[key].toLowerCase()
+  })
+}
+
+export default () => (req, res, next) => {
+  sanitizeAddresses(req.query)
```
Comment (sohkai) [inline packages/server/src/helpers/sanitize-address-middleware.js:11]: Ah, this is quite a cute middleware trick! Never thought of doing it like this, but it's a good way to dedupe the logic!

File: packages/server/src/helpers/sanitize-address-middleware.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { isAddress } from 'web3-utils'
+
+const sanitizeAddresses = object => {
+  Object.keys(object).forEach(key => {
```
Comment (sohkai) [inline packages/server/src/helpers/sanitize-address-middleware.js:4]: We could likely simplify this via `Object.entries()`, which will give us both the keys and values

File: packages/server/src/helpers/sanitize-address-middleware.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    if (!!object[key] && typeof object[key] === 'object') sanitizeAddresses(object[key])
+    else if(isAddress(object[key])) object[key] = object[key].toLowerCase()
```
Comment (sohkai) [inline packages/server/src/helpers/sanitize-address-middleware.js:6]: I would do a stricter check here—let's transform to checksum and error if that fails before passing it through as all lowercased.

Comment (sohkai) [reply packages/server/src/helpers/sanitize-address-middleware.js:6]: Ah, actually I think `isAdress()` checks for the checksum already right? In that case, ignore me :).

File: packages/server/src/helpers/sanitize-address-middleware.js (hunk 2 above)
Comment (facuspagnuolo) [reply packages/server/src/helpers/sanitize-address-middleware.js]: definitely!

File: packages/server/src/helpers/sanitize-address-middleware.js (hunk 3 above)
Comment (facuspagnuolo) [reply packages/server/src/helpers/sanitize-address-middleware.js]: yes it does :)

**Pull request merged**